docs/lib/restapi/fragments/native_common/authz/common.md

```diff
@@ -0,0 +1,124 @@
+```json
+{
+    "CognitoUserPool": {
+        "Default": {
+            "PoolId": "POOL-ID",
+            "AppClientId": "APP-CLIENT-ID",
+            "AppClientSecret": "APP-CLIENT-SECRET",
+            "Region": "us-east-1"
+        }
+    },
+    "CredentialsProvider": {
+        "CognitoIdentity": {
+            "Default": {
+                "PoolId": "YOUR-COGNITO-IDENTITY-POOLID",
+                "Region": "us-east-1"
+            }
+        }
+    }
+}
+```
+
+This will automatically add the Authorization header with access token to your requests to this API. To see more details on how to configure the API Gateway with the custom authorization, see [this](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-integrate-with-cognito.html)
+
+
+## Custom Interceptor
+
+If you have a custom token from another provider, you can set it `custom` and create your own interceptor
+
+```
+CUSTOM INTERCEPTOR CODE
+```
+
+## Note related to use Access Token or ID Token
+
+The ID Token contains claims about the identity of the authenticated user such as name, email, and phone_number. It could have custom claims as well, for example using [Amplify CLI](https://docs.amplify.aws/cli/usage/lambda-triggers#override-id-token-claims). On the Amplify Authentication category you can retrieve the Id Token using: 
+
+```javascript
+(await Auth.currentSession()).getIdToken().getJwtToken();
+``` 
+
+The Access Token contains scopes and groups and is used to grant access to authorized resources. [This is a tutorial for enabling custom scopes.](https://aws.amazon.com/premiumsupport/knowledge-center/cognito-custom-scopes-api-gateway/). You can retrieve the Access Token using 
+
+```javascript
+(await Auth.currentSession()).getAccessToken().getJwtToken();
+```
+
+## Customizing HTTP request headers
+
+To use custom headers on your HTTP request, you need to add these to Amazon API Gateway first. For more info about configuring headers, please visit [Amazon API Gateway Developer Guide](http://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-cors.html)
+
+If you have used Amplify CLI to create your API, you can enable custom headers by following above steps:  
+
+1. Visit [Amazon API Gateway console](https://aws.amazon.com/api-gateway/).
+3. On Amazon API Gateway console, click on the path you want to configure (e.g. /{proxy+})
+4. Then click the *Actions* dropdown menu and select **Enable CORS**
+5. Add your custom header (e.g. my-custom-header) on the text field Access-Control-Allow-Headers, separated by commas, like: 'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token,my-custom-header'.
+6. Click on 'Enable CORS and replace existing CORS headers' and confirm.
+7. Finally, similar to step 3, click the Actions drop-down menu and then select **Deploy API**. Select **Development** on deployment stage and then **Deploy**. (Deployment could take a couple of minutes).
```

Comment:
Do we need these, or can we just reference API Gateway's documentation, only?

Reply:
I actually pulled these from JS docs so we're having shared documentation, except native is under native_common folder/authz.md. At this point, i'd prefer to keep/remove this section when we consolidate the native_common with js